src/processor/operator/persistent/rel_batch_insert.cpp

```diff
@@ -53,6 +53,7 @@ void RelBatchInsert::executeInternal(ExecutionContext* context) {
     while (true) {
         relLocalState->nodeGroupIdx =
             partitionerSharedState->getNextPartition(relInfo->partitioningIdx);
```

Comment:
Can u pass `progressSharedState` into `partitionerSharedState->getNextPartition`? so you don't need to take another lock inside the `progressSharedState` to sync on `partitionsDone`. 

```
partition_idx_t PartitionerSharedState::getNextPartition(idx_t partitioningIdx,
    RelBatchInsertProgressSharedState& progress) {
    std::unique_lock xLck{mtx};
    if (nextPartitionIdx >= numPartitions[partitioningIdx]) {
        return INVALID_PARTITION_IDX;
    }
    progress.partitionsDone++;
    return nextPartitionIdx++;
}
```

Reply:
This ran into some issues because it would require `partitioner.h` to include `rel_batch_insert.h`, I changed it by having `RelBatchInsertProgressSharedState` in its separate file, is that alright?